tests/integration/adc/main.cpp

```diff
@@ -0,0 +1,354 @@
+#include <stdint.h>
+
+#include "../../../mcal/inc/Atmega32.h"
+#include "../../../utils/inc/Register.h"
+#include "../../../mcal/inc/Gpio.h"
+#include "../../../mcal/inc/Adc.h"
+#include "../../../utils/inc/Test.h"
+ 
+using namespace utils;
+using namespace avr::mcal::gpio;
+using namespace avr::types;
+using namespace avr::mcal::adc;
+
+int main () {
+    ADC.SelectChannel<ChannelMode::kSingleEnded_ADC0>();
```

Comment:
It will be a good practice if we use a separate scopes in order to separate between test cases. e.g.
```
{
    // TEST-CASE1
    // Configure ADC 1
   //  Print some data1
}
{
   // TEST-CASE2
   // Configure ADC 2
   //  Print some data2
}
```